src/scripts.rs

```diff
@@ -170,27 +163,17 @@ pub fn unvault_descriptor<Pk: MiniscriptKey>(
     managers_threshold: usize,
     cosigners: Vec<Pk>,
     csv_value: u32,
-) -> Result<UnvaultDescriptor<Pk>, Error> {
+) -> Result<UnvaultDescriptor<Pk>, ScriptCreationError> {
     if stakeholders.is_empty() || managers.is_empty() || cosigners.len() != stakeholders.len() {
-        return Err(Error::ScriptCreation(
-            "Unvault: bad parameters. There must be a non-zero \
-                number of managers and non_managers, and as many cosigners as non_managers"
-                .to_string(),
-        ));
+        return Err(ScriptCreationError::BadParameters);
     }
 
     if managers_threshold > managers.len() {
-        return Err(Error::ScriptCreation(
-            "Unvault: bad parameters. The managers threshold is higher than \
-                the number of managers."
-                .to_owned(),
-        ));
+        return Err(ScriptCreationError::BadParameters);
     }
 
     if (csv_value & (1 << 22)) != 0 {
```

Comment:
since error is the same maybe do:
```rust
if stakeholders.is_empty() || managers.is_empty() || cosigners.len() != stakeholders.len() ||
    managers_threshold > managers.len() || 
    (csv_value & (1 << 22)) != 0 {
        return Err(ScriptCreationError::BadParameters);
    }
```

Reply:
I'm hesitant: it really overloads the `if` and makes it less readable imho